### PR TITLE
Update contrib docs

### LINKED ISSUE #5125: Update building.md for Bindgen
The building instructions for the bindgen branch is not up to date. Needs fix.

### PR DESCRIPTION
**Pull request by elle-j**
## What, How & Why?

As a lot of our information in the `contrib/` docs are outdated after the v12 rewrite, this PR updates some of this information. Changes here pertain mostly to building Realm JS and debugging C++.

#### TODOs

Left some `TODO`s in this first walkthru of the docs as it could be good to get a second set of eyes 🔍 and input on those parts, either as updates here or as follow-ups.

#### Files not modified in this PR:

* We can likely remove [contrib/guide-testing-with-sample-apps.md](https://github.com/realm/realm-js/blob/main/contrib/guide-testing-with-sample-apps.md) as we have [contrib/debug-rn-examples.md](https://github.com/realm/realm-js/blob/main/contrib/debug-rn-examples.md).
* [contrib/debugging-react-native.md](https://github.com/realm/realm-js/blob/main/contrib/debugging-react-native.md) needs updating or perhaps combine some of it with `contrib/debug-rn-examples.md` after making necessary updates.
* [guide-memory-leaks.md](https://github.com/realm/realm-js/blob/main/contrib/guide-memory-leaks.md)
* [guide-watchman.md](https://github.com/realm/realm-js/blob/main/contrib/guide-watchman.md)
* README's in `integration-tests/`.

This closes #5125

### CODE REVIEW

File: contrib/building.md
Comment (kraenhansen) [inline contrib/building.md:170]: Consider putting these in markdown comments? Or do you plan to resolve these before merging?

File: contrib/building.md
Comment (kraenhansen) [inline contrib/building.md:187]:
```suggestion
> If you have cloned the repo previously make sure you remove your `node_modules` directory, as well as any `node_modules` directory of any sub-directory, since it may contain stale dependencies which may cause the build to fail.
```

File: contrib/debugging-cpp.md
Comment (kraenhansen) [inline contrib/debugging-cpp.md:45]: Hmm ... 🤔 I can't say why that'd happen, on the top of my head.

File: contrib/building.md
Comment (kraenhansen) [inline contrib/building.md:198]: If you want to be really fancy 😄 
```suggestion
2. Press <kbd>Shift</kbd>+<kbd>Cmd</kbd>+<kbd>P</kbd> to open the command palette
```

File: contrib/logger/sync_logging.md
Comment (kraenhansen) [inline contrib/logger/sync_logging.md:12]: This line seems outdated (as it mentions `js_sync.hpp`. We might just want to delete the entire file, since this should now be controllable through the `syncLogLevel` context variable as outlined here: https://github.com/realm/realm-js/tree/main/integration-tests#setting-context

Also, this might change when / if we merge https://github.com/realm/realm-js/pull/6305

**Review by kraenhansen — APPROVED** [review]
So many improvements! It's been long overdue on my part to go through this. Thank you so, so, much for taking the time to dig into this! 💙

I'd say this as so many improvements that it could merge right away, if you just move the TODOs into markdown comments 👍

File: contrib/building.md
Comment (elle-j) [reply contrib/building.md:170]: I'll go ahead and update those to MD comments, it was mainly to make it hard to miss for reviewers. I'll leave it in for an upcoming iteration that we can look at when people are back from the holidays 👍 🎅

File: contrib/building.md
Comment (elle-j) [reply contrib/building.md:198]: Oh I want to be fancy 😌

File: contrib/logger/sync_logging.md
Comment (elle-j) [reply contrib/logger/sync_logging.md:12]: You're right, must have missed that last sentence. Deleting the file seems reasonable. Regarding #6305, once that gets merged (or in that same PR), we can just update the context variable 👍

**Pull request merged**
